driver/bpf/plumbing_helpers.h

```diff
@@ -417,6 +423,34 @@ static __always_inline int bpf_test_bit(int nr, unsigned long *addr)
 	return 1UL & (_READ(addr[BIT_WORD(nr)]) >> (nr & (BITS_PER_LONG - 1)));
 }
 
+#ifdef CAPTURE_SCHED_PROC_FORK
+static __always_inline bool drop_syscall_child_events(void *ctx, enum ppm_event_type evt_type)
+{
+	switch (evt_type) {
+	case PPME_SYSCALL_CLONE_20_X:
+	case PPME_SYSCALL_FORK_20_X:
+	case PPME_SYSCALL_VFORK_20_X:
+	case PPME_SYSCALL_CLONE3_X: {
+		/* On s390x, clone and fork child events will be generated but
+		 * due to page faults, no args/envp information will be collected.
+		 * Also no child events appear for clone3 syscall.
+		 *
+		 * Because child events are covered by CAPTURE_SCHED_PROC_FORK,
+		 * let proactively ignore them.
+		 */
```

Comment:
Uhm so without this patch I expect actual tests are failing on s390x with the old probe, is it true? Test are asserting events absence but if these events are generated tests should fail :thinking:

```cpp
#ifdef CAPTURE_SCHED_PROC_FORK
	evt_test->assert_event_absence(ret_pid);
#else
	evt_test->assert_event_presence(ret_pid);
```

What about the kmod, do we have to put in place the same logic?

Reply:
Hi @Andreagit97 ,

> Uhm so without this patch I expect actual tests are failing on s390x with the old probe, is it true? Test are asserting events absence but if these events are generated tests should fail thinking

That's true:
````
[ RUN      ] SyscallExit.cloneX_child
/root/git/falcosecurity/libs/test/drivers/event_class/event_class.cpp:921: Failure
Failed
There is an event '223' in the buffers, but it shouldn't be there

[  FAILED  ] SyscallExit.cloneX_child (92 ms)
````
That's why I used the similar "shortcut" as you did for modern BPF some time back -- as those child events has been seen there as well -- read [here](https://github.com/falcosecurity/libs/pull/648#issuecomment-1288679138)

> What about the kmod, do we have to put in place the same logic?

I have not yet tried. But last try on that topic was [here](https://github.com/falcosecurity/libs/pull/648#issuecomment-1289171826) -- not seen w/ kmod.

Reply:
yes, I remember the pain related to this `clone/fork` child issue. Probably we need to disable this logic also in the kmod just to have at least feature parity, moreover, I've seen that I missed this logic in the modern bpf `vfork` :face_exhaling: I will do that in another PR, thank you for spotting this incongruence!

Reply:
You are welcome.

Running `driver_test` against the `kmod` does not produce those child events. Also here the good info is that all 243 tests run successfully.

Reply:
> Running driver_test against the kmod does not produce those child events

yeah same on x86, but probably it makes sense to follow the same workaround between our 3 drivers